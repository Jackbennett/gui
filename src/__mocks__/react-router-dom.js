import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
// eslint-disable-next-line no-import-assign
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
