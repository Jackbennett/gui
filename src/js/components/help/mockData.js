export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
