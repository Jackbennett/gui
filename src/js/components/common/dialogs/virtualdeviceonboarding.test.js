import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import VirtualDeviceOnboarding from './virtualdeviceonboarding';
import { undefineds } from '../../../../../tests/mockData';

const mockStore = configureStore([thunk]);

describe('VirtualDeviceOnboarding Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      app: { features: { isHosted: false } },
      users: {
        organization: { tenant_token: null }
      }
    });
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <VirtualDeviceOnboarding />
          </Provider>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree)).toEqual(expect.not.stringMatching(undefineds));
  });
});
