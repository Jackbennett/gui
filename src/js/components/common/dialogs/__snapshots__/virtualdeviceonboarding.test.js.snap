// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VirtualDeviceOnboarding Component renders correctly 1`] = `
<div>
  <div>
    <b>
      1. Prerequisites
    </b>
    <p>
      As you are running Mender on-premise, for these instructions we assume that you already have Docker installed and the Mender integration environment up and running on your machine.
    </p>
    <p>
      To start a virtual device, change directory into the folder where you cloned Mender integration.
    </p>
  </div>
  <p>
    <b>
      2. Copy & paste and run the following command to start the virtual device:
    </b>
  </p>
  <div
    class="code"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      style="float: right; margin: -20px 0px 0px 10px;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z"
            />
          </svg>
        </span>
        Copy to clipboard
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <span
      style="white-space: pre-wrap; word-break: break-word;"
    >
      ./demo --client up
    </span>
  </div>
  <p />
  <p>
    The device should appear in the Pending devices view in a couple of minutes.
  </p>
  <p>
    Visit
     
    <a
      href="https://docs.mender.io/development/get-started/preparation/prepare-a-virtual-device"
      rel="noopener noreferrer"
      target="_blank"
    >
      our documentation
    </a>
     
    for more information on managing the virtual device.
  </p>
</div>
`;
